package pyRpc2.pyRpc;

enum ErrorType {
  ET_BAD_REQUEST_DATA = 0;
  ET_BAD_REQUEST_PROTO = 1;
  ET_SERVICE_NOT_FOUND = 2;
  ET_METHOD_NOT_FOUND = 3;
  ET_RPC_ERROR = 4;
  ET_RPC_FAILED = 5;

  ET_INVALID_REQUEST_PROTO = 6;
  ET_BAD_REPLY_PROTO = 7;
  ET_UNKNOWN_HOST = 8;
  ET_IO_ERROR = 9;
}

message Request {
  required string service_name = 1;
  required string method_name = 2;
  required bytes request_proto = 3;
}

message Reply {
  optional bytes reply_proto = 1;
  optional string error = 2;
  optional bool callback = 3 [default = false];
  optional ErrorType err_reason = 4;
}
